zcncore/smartcontract_executor_mobile.go

```diff
@@ -0,0 +1,31 @@
+//go:build mobile
+// +build mobile
+
+package zcncore
+
+type SmartContractExecutor interface {
+	// ExecuteSmartContract implements wrapper for smart contract function
+	ExecuteSmartContract(address, methodName string, input string, val uint64) error
```

Comment:
Why do we differ the signature, if this breaks the mobile, then we should let the mobile code updated to adjust the changes.

Reply:
At present, only a subset of Go types are supported. https://pkg.go.dev/golang.org/x/mobile/cmd/gobind

Reply:
All exported symbols in the package must have types that are supported. Supported types include:

- Signed integer and floating point types.

- String and boolean types.

- Byte slice types. Note that byte slices are passed by reference,
  and support mutation.

- Any function type all of whose parameters and results have
  supported types. Functions must return either no results,
  one result, or two results where the type of the second is
  the built-in 'error' type.

- Any interface type, all of whose exported methods have
  supported function types.

- Any struct type, all of whose exported methods have
  supported function types and all of whose exported fields
  have supported types.

Reply:
okay, got it.